More flexible database (#80)
ansible/awx-operator: README.md

Again, this is the most relaxed SCC that is provided by OpenShift, so be sure to familiarize yourself with the security concerns that accompany this action.

### Database Setup

The AWX Operator supports two different scenarios with regard to database setup.

> One can either let the Operator manages a PostgreSQL deployment within the same namespace the AWX deployment will take place. Or, one can connect to an already - externally managed - PostgreSQL instance. The decision is made based on the boolean value of the `external_database` ansible variable.

Reviewer: ```suggestion
One can either let the Operator manage a PostgreSQL deployment within the same namespace as AWX, or one can connect to an externally managed PostgreSQL instance. The decision is made based on the boolean value of the `external_database` Ansible variable.
````

#### External Database

In the scenario when one wants to rely on an external PostgreSQL instance, one needs to provide the necessary information to the operator.

There are three ways one can specify those information. Those way are listed in order of priority in which they are treated.

* Providing a specifically crafted secret

> One can create a specially formated secret within the namespace the AWX instance will be deployed into. This secret needs then to be specified as `tower_postgres_configuration_secret` variable in the spec requirements.

Reviewer: ```suggestion
One can create a specially formated `Secret` within the namespace the AWX instance will be deployed into. This `Secret` needs then to be specified as `tower_postgres_configuration_secret` variable under the `spec` field in your AWX Custom Resource.
```

  tower_postgres_configuration_secret: my-awx-external-db-secret
```

The secret format is the following:

```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: <name-of-the-secret>
  namespace: <namespace>
stringData:
  password: <password>
  username: <username>
  database: <database>
  port: <port>
  host: <host>
```

* Providing variables (not recommended)

If one does not wish to use secret, one can specify the following variables in the spec requirements.

| Variable                | Descritpion                       | Default Value |
| ----------------------- | --------------------------------- | ------------- |
| tower_postgres_user     | Username to connect to PostgreSQL |               |
| tower_postgres_pass     | Password to connect to PostgreSQL |               |
| tower_postgres_database | Database name to connect to       |               |
| tower_postgres_host     | Host to connect to PostgreSQL     |               |
| tower_postgres_port     | Port to connect to PostgreSQL     | 5432          |

The following is an example of an AWX spec section that relie on those variables.

```yaml
spec:
  external_database: true
  tower_postgres_user: awx

>   tower_postgres_pass: awx

Reviewer: ⚠️ This is not secure. We should _never_ instruct the user to provide sensitive information in anything except a `Secret`.
